spring-security-modules/spring-security-web-boot-4/src/main/java/com/baeldung/securityfilterchain/configuration/SecurityConfig.java

```diff
@@ -2,49 +2,42 @@
+import org.springframework.context.annotation.Configuration;
 import org.springframework.http.HttpMethod;
-import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityCustomizer;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
 import org.springframework.security.config.http.SessionCreationPolicy;
 import org.springframework.security.web.SecurityFilterChain;
 
+@Configuration
 @EnableWebSecurity
-@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true, jsr250Enabled = true)
+@EnableMethodSecurity(securedEnabled = true, jsr250Enabled = true)
 public class SecurityConfig {
 
     @Value("${spring.security.debug:false}")
     boolean securityDebug;
 
     @Bean
     public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-        http.csrf()
-          .disable()
-          .authorizeRequests()
-          .antMatchers(HttpMethod.DELETE)
-          .hasRole("ADMIN")
-          .antMatchers("/admin/**")
-          .hasAnyRole("ADMIN")
-          .antMatchers("/user/**")
-          .hasAnyRole("USER", "ADMIN")
-          .antMatchers("/login/**")
-          .permitAll()
-          .anyRequest()
-          .authenticated()
-          .and()
-          .httpBasic()
-          .and()
-          .sessionManagement()
-          .sessionCreationPolicy(SessionCreationPolicy.STATELESS);
+        http.csrf(AbstractHttpConfigurer::disable)
+          .authorizeHttpRequests(authorizationManagerRequestMatcherRegistry ->
+                  authorizationManagerRequestMatcherRegistry.requestMatchers(HttpMethod.DELETE).hasRole("ADMIN")
+                          .requestMatchers("/admin/**").hasAnyRole("ADMIN")
+                          .requestMatchers("/user/**").hasAnyRole("USER", "ADMIN")
+                          .requestMatchers("/login/**").permitAll()
+                          .anyRequest().authenticated())
+          .httpBasic(Customizer.withDefaults())
+          .sessionManagement(httpSecuritySessionManagementConfigurer -> httpSecuritySessionManagementConfigurer.sessionCreationPolicy(SessionCreationPolicy.STATELESS));
 
         return http.build();
     }
 
     @Bean
     public WebSecurityCustomizer webSecurityCustomizer() {
-        return (web) -> web.debug(securityDebug)
-          .ignoring()
-          .antMatchers("/css/**", "/js/**", "/img/**", "/lib/**", "/favicon.ico");
+        return web -> web.debug(securityDebug).ignoring().requestMatchers("/css/**", "/js/**", "/img/**", "/lib/**", "/favicon.ico");
```

Comment:
Can you check the formatting here pls, this line seems a bit too long.

Reply:
I have a sonar lint notification that warns me to override also the equals and hashCode since the methods are overridden in the super class. We can remove it.

Reply:
Done